prov/efa/src/efa_mr.c

```diff
@@ -227,7 +221,32 @@ static int efa_mr_hmem_setup(struct efa_mr *efa_mr,
 		efa_mr->needs_sync = true;
 		efa_mr->peer.device.cuda = attr->device.cuda;
 
-		if (cuda_is_gdrcopy_enabled()) {
+		if ((flags & FI_MR_DMABUF) != 0) {
+			if (!cuda_is_dmabuf_supported()) {
+				/*
+				 * caller allegedly is passing us a valid file descriptor
+				 * exported from the nvidia driver, but we directly queried the
+				 * cuda property for dmabuf export support during init, and
+				 * support was explicitly returned as false. Refuse to service
+				 * the request.
+				 */
+				return -FI_EOPNOTSUPP;
+			}
+			if (!efa_mr->domain->hmem_info[FI_HMEM_CUDA].p2p_supported_by_device) {
+				/*
+				 * To get here, we must have previously failed to export a
+				 * dmabuf handle from a CUDA ourselves (see
+				 * efa_domain_hmem_info_check_p2p_support_cuda), but it must
+				 * also be the case that we've queried dmabuf export support
+				 * property during init and cuda told us that the driver is a
+				 * valid exporter. Warn once, optimistically assume that the
+				 * user succeeded despite our earlier failure, and proceed.
+				 */
+				EFA_WARN_ONCE(FI_LOG_MR,
```

Comment:
This shouldn't be only a warning. In L850 we will skip IB registration if p2p_supported_by_device is false, which is undesirable for NCCL plugin on p5 series (because you requested FI_OPT_CUDA_API_PERMITTED in setopt as false, we will make that setopt fail if `p2p_supported_by_device` is false.